src/node_rc.cc

```diff
@@ -0,0 +1,127 @@
+        return it->substr(flag.size() + 1);
+      }
+    }
+  }
+
+  return std::nullopt;
+}
+
+ConfigReader::ParseResult ConfigReader::ParseConfig(
+    const std::string& config_path) {
+  std::string file_content;
+  // Read the configuration file
+  int r = ReadFileSync(&file_content, config_path.c_str());
+  if (r != 0) {
+    const char* err = uv_strerror(r);
+    FPrintF(
+        stderr, "Cannot read configuration from %s: %s\n", config_path, err);
+    return ParseResult::FileError;
+  }
+
+  // Parse the configuration file
+  simdjson::ondemand::parser json_parser;
+  simdjson::ondemand::document document;
+  if (json_parser.iterate(file_content).get(document)) {
+    FPrintF(stderr, "Can't parse %s\n", config_path.c_str());
+    return ParseResult::InvalidContent;
+  }
+
+  simdjson::ondemand::object main_object;
+  // If document is not an object, throw an error.
+  if (auto root_error = document.get_object().get(main_object)) {
+    if (root_error == simdjson::error_code::INCORRECT_TYPE) {
+      FPrintF(stderr,
+              "Root value unexpected not an object for %s\n\n",
+              config_path.c_str());
+    } else {
+      FPrintF(stderr, "Can't parse %s\n", config_path.c_str());
+    }
+    return ParseResult::InvalidContent;
+  }
+
+  ConfigReader::Config config;
+  simdjson::ondemand::value ondemand_value;
+  simdjson::ondemand::raw_json_string key;
+
+  for (auto field : main_object) {
+    if (field.key().get(key) || field.value().get(ondemand_value)) {
+      return ParseResult::InvalidContent;
+    }
+    if (key == "experimental_transform_types") {
+      if (ondemand_value.get_bool().get(config.experimental_transform_types)) {
+        FPrintF(stderr, "Invalid value for experimental_transform_types\n");
+        return ParseResult::InvalidContent;
+      }
+    }
+
+    if (key == "import") {
+      simdjson::ondemand::array raw_imports;
+      if (ondemand_value.get_array().get(raw_imports)) {
+        FPrintF(stderr, "Invalid value for import\n");
+        return ParseResult::InvalidContent;
+      }
+      for (auto raw_import : raw_imports) {
+        std::string_view import;
+        if (raw_import.get_string(import)) {
+          FPrintF(stderr, "Invalid value for import\n");
+          return ParseResult::InvalidContent;
+        }
+        config.import.push_back(std::string(import));
+      }
+    }
+  }
+
+  config_ = config;
+  return ParseResult::Valid;
+}
+
+void ConfigReader::AssignNodeOptions(std::string* node_options) {
+  std::vector<std::string> result;
+  if (config_.experimental_transform_types) {
+    result.push_back("--experimental-transform-types");
+  }
+
+  if (config_.import.size() > 0) {
+    for (const auto& import : config_.import) {
+      result.push_back("--import=" + import);
+    }
+  }
+  if (result.empty()) {
+    *node_options = "";
+  } else {
+    *node_options = result[0];
+    for (size_t i = 1; i < result.size(); ++i) {
+      *node_options += " " + result[i];
```

Comment:
This is a bit dangerous since someone could do something like
```json
"import": [
 "./foo.js --some-other-flag"
]
```
And the other flag will be evaluated 🤔